core/src/services/gcs/core.rs

```diff
@@ -149,19 +155,32 @@ impl GcsCore {
             write!(&mut url, "&predefinedAcl={}", acl).unwrap();
         }
 
-        let mut req = Request::post(&url);
+        let mut req =
+            Request::post(&url).header(CONTENT_TYPE, "multipart/related; boundary=my-boundary");
```

Comment:
I don't feel comfortable with handwriting multipart request. How about using `request` multipart support? We can extend `AsyncBody`'s Multipart to allow multiple fields.

Example could be found at https://github.com/apache/incubator-opendal/blob/main/core/src/services/ipmfs/writer.rs#L44